### DISCUSSION TITLE
Catch the click in a button contained in an expansion header

### DISCUSSION THREAD

**Discussion by superlou** [zauberzeug/nicegui#1554] [Q&A] [answered]
### Question

I am trying to include a button in the header slot of an expansion element, e.g.:

```py
with ui.expansion() as expansion:
    with expansion.add_slot("header"):
        with ui.row():
            ui.label("Expansion Header Text")
            ui.button("Test", on_click=lambda: print("here"))
```

However, when the button is clicked, the expansion expands/contracts with each click, when I just want it to run the button's `on_click` method. Is there a way to catch the click in the button only?

Edit: There's the `expand-icon-toggle` prop which will make the expansion only toggle when clicking the expansion icon, but that's less convenient.

**Comment by rodja** [comment, marked as answer]
You can use the [`.stop` event modifier](https://vuejs.org/guide/essentials/event-handling.html#event-modifiers) from Vue:

```py
with ui.expansion() as expansion:
    with expansion.add_slot("header"):
        with ui.row():
            ui.label("Expansion Header Text")
            ui.button("Test").on('click.stop', lambda: print("here"))
```

**Reply by superlou** [reply]
That works, thanks!
